src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/SHA256.cs

```diff
@@ -36,21 +30,22 @@ public Implementation()
                 HashSizeValue = _hashProvider.HashSizeInBytes * 8;
             }
 
-            protected sealed override void HashCore(byte[] array, int ibStart, int cbSize)
-            {
+            protected sealed override void HashCore(byte[] array, int ibStart, int cbSize) =>
                 _hashProvider.AppendHashData(array, ibStart, cbSize);
-            }
 
-            protected sealed override byte[] HashFinal()
-            {
-                return _hashProvider.FinalizeHashAndReset();
-            }
+            protected override void HashCore(ReadOnlySpan<byte> source) =>
```

Comment:
We'll need to agree to disagree then.  The rest of corefx does not do what you're proposing, and we should be consistent with that.